### ISSUE TITLE
Is cross validation only testing extrapolative qualities?  (TimeSeriesSplit)

### ISSUE THREAD

**Issue by Shmuppel** [johanneskopton/ffist#5]
Hiya! 

First of all thanks for the great lib, not often you find something that does exactly what you're looking for. Also always fun to to see Sytze's name pop up (currently a student under him. My background is mostly neural networks and not so much Kriging so i'm not sure if I understand the processes completely, but I had a question regarding the cross validation. 

It looks like you're using TimeSeriesSplit, also with the note in the introductory collab: "we are only using observations from the past to infer to our new locations". By doing cross validation this way aren't we only testing the extrapolative qualities of our model? For my use case i'm mostly interested in interpolating, and the autocorrelation in the time dimension is rather short compared to the range of the dataset (i'm interpolating weather, and much less interested in extrapolation). 

Does the kriging account for this somehow in a way i'm not familiar with, or is this something I should look into? I wouldn't be surprised if my model performed worse if it had to predict unreasonably far into the future every time. 

Thanks again and kind regards,
Niels

**Comment by johanneskopton** [comment]
Hi Niels,

happy to hear that you are interested in this package and that it can be of help to you.

You are completely right, the cross validation is indeed only testing extrapolative qualities. My initial use-case was making live predictions every day, so we would not have future data. So, in order to reflect this while validating on historical data, I used TimeSeriesSplit. But you are right, I present this as a package for "Interpolation and Forecasting", so CV should be useful to also test interpolative qualities.

I will make the CV split customizable asap.

Oh, and say hi to Sytze! Loved working with him.

Cheers,
Johannes

**Comment by johanneskopton** [comment]
Hi Niels,

does this solve your problem? I am not sure, what the best split function would be for your use case. Maybe [shuffle](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.ShuffleSplit.html) or [leave-one-out](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.LeaveOneOut.html)?

Let me know if that helps or if there is anything to improve.

If this is fine, I will merge it to main and push to PyPI.

Cheers!

**Comment by Shmuppel** [comment]
Hi Johannes,

Thanks for the super quick response, that's awesome! I think something like this would be good enough for my use case, using the shuffle split. Leave one out would be a bit too costly for the size of my data, but this is a very nice way of still keeping that option in.

Kfold cross validator would be very useful too (which guarantees test sets are mutually exclusive, which shuffle doesn't), but then we'd have to pass an argument in order to shuffle the data (shuffle = True). One option is to pass an cv_split_kwargs argument so users can customize the cv_split arguments should they desire to. Either way should work fine for my usecase!

Regardless I can give it a go tomorrow and let you know how it goes. 

Many thanks!
Niels

**Comment by johanneskopton** [comment]
Ahh, good idea! You could also do this without further changes to the package like this:
```python
class ShuffledKFold(sklearn.model_selection.KFold):
    def __init__(self, n_splits):
        super().__init__(n_splits, shuffle=True)
```
And then pass this as `cv_split`. But maybe unnecessarily cumbersome.

Oh, and I also noticed, that I messed up the type hints. Its a class, rather than a callable. Will fix that tomorrow.

Cheers,
Johannes

**Comment by Shmuppel** [comment]
Ah great, wouldn't say its cumbersome per sé, adding an extra argument is also a bit bloaty (especially since the defaults do just fine for most people). This looks good to me!

**Comment by Shmuppel** [comment]
Hi Johannes,

Just wanted to let you know this solution works nicely, feel free to merge it (after the small type hint change)!
I got another point that may be of interest but i'll make a seperate issue for it.

Thanks again! 

<img width="555" alt="Image" src="https://github.com/user-attachments/assets/d2965a96-1369-4e31-90f3-23c1e07ec64c" />

**Comment by johanneskopton** [comment]
Thanks a lot for the feedback and sorry for the delay. Will merge it now 🙏
